src/webgl/interaction.js

```diff
@@ -1,107 +1,141 @@
+/**
+ * @module Lights, Camera
+ * @submodule Interaction
+ */
+
 'use strict';
 
 var p5 = require('../core/core');
 
 /**
- * Allows rotation of a 3D sketch by dragging the mouse. As the mouse is dragged
- * away from the center of the canvas in the X or Y direction, the sketch is
- * rotated about the Y or X axis respectively. Note that this rotation only
- * affects objects drawn after orbitControl() has been called in the draw() loop.
- * To reverse movement in either axis, enter a negative number for sensitivity.
- * Calling this function without arguments is equivalent to calling orbitControl(1,1).
+ * Allows movement around a 3D sketch using a mouse or trackpad.  Left-clicking
+ * and dragging will rotate the camera position about the center of the sketch,
+ * right-clicking and dragging will pan the camera position without rotation,
+ * and using the mouse wheel (scrolling) will move the camera closer or further
+ * from the center of the sketch. This function can be called with parameters
+ * dictating sensitivity to mouse movement along the X and Y axes.  Calling
+ * this function without parameters is equivalent to calling orbitControl(1,1).
+ * To reverse direction of movement in either axis, enter a negative number
+ * for sensitivity.
  * @method orbitControl
  * @for p5
- * @param  {Number} [sensitivityX]        sensitivity to mouse movement along X axis
- * @param  {Number} [sensitivityY]        sensitivity to mouse movement along Y axis
+ * @param  {Number} [sensitivityX] sensitivity to mouse movement along X axis
+ * @param  {Number} [sensitivityY] sensitivity to mouse movement along Y axis
  * @chainable
  * @example
  * <div>
  * <code>
  * function setup() {
  *   createCanvas(100, 100, WEBGL);
+ *   normalMaterial();
  * }
- *
  * function draw() {
  *   background(200);
- *   // Orbit control allows the camera to orbit around a target.
  *   orbitControl();
+ *   rotateY(0.5);
  *   box(30, 50);
  * }
  * </code>
  * </div>
  *
  * @alt
- * Camera orbits around box when mouse is hold-clicked & then moved.
+ * Camera orbits around a box when mouse is hold-clicked & then moved.
  */
-//@TODO: implement full orbit controls including
-//pan, zoom, quaternion rotation, etc.
+
 // implementation based on three.js 'orbitControls':
 // https://github.com/mrdoob/three.js/blob/dev/examples/js/controls/OrbitControls.js
 p5.prototype.orbitControl = function(sensitivityX, sensitivityY) {
   this._assert3d('orbitControl');
   p5._validateParameters('orbitControl', arguments);
 
+  // If the mouse is not in bounds of the canvas, disable all behaviors:
+  var mouseInCanvas =
+    this.mouseX < this.width &&
+    this.mouseX > 0 &&
+    this.mouseY < this.height &&
+    this.mouseY > 0;
```

Comment:
hmm, how does this feel in use in the web editor? It's very common to have the mouse in focus but of out frame there. I wonder if it will feel totally fine, or a little discontinuous?

Reply:
I added this check to avoid page-scrolling activating the zoom.  In that case, if a canvas were halfway down a page, the camera view would be too far from the origin to see the sketch by the time the user reached the canvas.  But perhaps it is not necessary for panning or zooming.  [Here](http://alpha.editor.p5js.org/full/BkqOcCrHm) is a web-editor example with two p5 library files: 
* "p5.js" is the current behavior (mouse position checked for orbit/pan/zoom)
* "p5-mousecheckonzoom.js" mouse position only checked for zoom

In the 2nd case, right-clicking outside of the canvas still activates the context menu (only deactivated on the canvas element itself), stopping the panning behavior anyway.  But a user can start a pan with the mouse inside of the canvas, and continue it with the mouse outside of the canvas, which seems correct.   

This case also introduces one odd behavior:  if you right-click and hold outside of the canvas, then release the mouse inside the canvas, the context-menu is disabled and panning is activated despite the mouse no longer being pressed.  This lasts until you click the mouse again.  I'm not sure that this is a common enough / weird enough behavior to constitute a bug, but I just wanted to point it out.

I think this 2nd behavior probably makes the most sense.  Thoughts?

Reply:
Cool, thank you for putting this example together! I agree that it's fine-as-is, with the stops. And as for the weird bug, I think it's ok to file that as an edge case issue and see if we can develop a solution for it in the future? I think you're right that it's unlikely to occur, but good to have documented just in case.

Reply:
@kadamwhite @AidanNelson 
This is causing an annoying needs of first Touch on Mobile in order to activate orbitControl on a Phone (or touch device) 

I already create an issue for this. But not sure which will be the best approach
https://github.com/processing/p5.js/issues/5176